common/p_setup.cpp

```diff
@@ -870,14 +870,44 @@ void P_LoadLineDefs (const int lump)
 	memset (lines, 0, numlines*sizeof(line_t));
 	data = (byte *)W_CacheLumpNum (lump, PU_STATIC);
 
+	// [Blair] Don't mind me, just hackin'
+	// E2M7 has flags masked in that interfere with MBF21 flags.
+	// Boom fixes this with the comp flag comp_reservedlineflag
+	// We'll fix this for now by just checking for the E2M7 FarmHash
+	const char e2m7hash[33] = "43ffa244f5ae923b7df59dbf511c0468";
+
+	char levelHash[33];
+
+	ArrayInit(levelHash, '0');
+
+	// [Blair] Serialize the hashes before reading.
+	uint64_t reconsthash1 = (uint64_t)(::level.level_fingerprint[0]) |
+	                        (uint64_t)(::level.level_fingerprint[1]) << 8 |
+	                        (uint64_t)(::level.level_fingerprint[2]) << 16 |
+	                        (uint64_t)(::level.level_fingerprint[3]) << 24 |
+	                        (uint64_t)(::level.level_fingerprint[4]) << 32 |
+	                        (uint64_t)(::level.level_fingerprint[5]) << 40 |
+	                        (uint64_t)(::level.level_fingerprint[6]) << 48 |
+	                        (uint64_t)(::level.level_fingerprint[7]) << 56;
+
+	uint64_t reconsthash2 = (uint64_t)(::level.level_fingerprint[8]) |
+	                        (uint64_t)(::level.level_fingerprint[9]) << 8 |
+	                        (uint64_t)(::level.level_fingerprint[10]) << 16 |
+	                        (uint64_t)(::level.level_fingerprint[11]) << 24 |
+	                        (uint64_t)(::level.level_fingerprint[12]) << 32 |
+	                        (uint64_t)(::level.level_fingerprint[13]) << 40 |
+	                        (uint64_t)(::level.level_fingerprint[14]) << 48 |
+	                        (uint64_t)(::level.level_fingerprint[15]) << 56;
+
+	sprintf(levelHash, "%16llx%16llx", reconsthash1, reconsthash2);
```

Comment:
Please do not use `sprintf`.

Additionally, I don't think that `ll` is compatible with C++98.  I would recommend using pstdint.h's `PRINTF_INT64_MODIFIER`.

Reply:
Changed to use StrFormat.

llx isn't C++98 but then again, neither is uint64_t. They're both C99 and C++11. I've changed this to the PRIx64 macro (pstdint.h was trying to redefine a few types).